$OpenBSD: patch-src_posix_thread_c,v 1.5 2021/04/03 11:15:46 sthen Exp $

- Make use of clock_gettime() on OpenBSD. Until we have per-process
  timers and can enable _POSIX_TIMERS in unistd.h.
- Make use of _SC_NPROCESSORS_ONLN.

Index: src/posix/thread.c
--- src/posix/thread.c.orig
+++ src/posix/thread.c
@@ -62,7 +62,7 @@
 # define _POSIX_MONOTONIC_CLOCK (-1)
 #endif
 
-#if (_POSIX_TIMERS > 0)
+#if (_POSIX_TIMERS > 0) || defined(__OpenBSD__)
 static unsigned vlc_clock_prec;
 
 # if (_POSIX_MONOTONIC_CLOCK > 0) && (_POSIX_CLOCK_SELECTION > 0)
@@ -625,7 +625,7 @@ void vlc_control_cancel (int cmd, ...)
 
 mtime_t mdate (void)
 {
-#if (_POSIX_TIMERS > 0)
+#if (_POSIX_TIMERS > 0) || defined(__OpenBSD__)
     struct timespec ts;
 
     vlc_clock_setup ();
@@ -712,6 +712,8 @@ unsigned vlc_GetCPUCount(void)
         count = sysconf (_SC_NPROCESSORS_ONLN);
     free (cpulist);
     return count ? count : 1;
+#elif defined(_SC_NPROCESSORS_ONLN)
+    return sysconf(_SC_NPROCESSORS_ONLN);
 #elif defined(_SC_NPROCESSORS_CONF)
     return sysconf(_SC_NPROCESSORS_CONF);
 #else
